ldms/src/core/ldms_xprt.c

```diff
@@ -3764,20 +3764,24 @@ int ldms_xprt_sockaddr(ldms_t _x, struct sockaddr *local_sa,
 {
 	struct ldms_xprt *x = _x;
 	zap_err_t zerr;
- again:
-	if (x->sa_len) {
-		/* use cached addresses */
-		*sa_len = x->sa_len;
-		memcpy(local_sa, &x->local_sa, x->sa_len);
-		memcpy(remote_sa, &x->remote_sa, x->sa_len);
-		return 0;
-	}
-	zerr = zap_get_name(x->zap_ep, &x->local_sa, &x->remote_sa, &x->sa_len);
```

Comment:
@tom95858 `ldms_xprt_sockaddr` pass `&x->sa_len` instead of `sa_len`. `x->sa_len` is always zero at this line. This is a bug.

Reply:
@nichamon, on input to these address functions, sa_len is the size of the buffer. On output sa_len is the size of the address written to the buffer. It should be something like this:
```
socklen_t sa_len = sizeof(x->local_sa);
zap_get_name(..., &x->local_sa, &sa_len);
x->sa_len = sa_len;
```